src/devices/Units/Pressure.cs

```diff
@@ -120,51 +120,110 @@ public static Pressure FromMillimeterOfMercury(double value)
             return new Pressure(value / MillimeterOfMercuryRatio);
         }
 
+        /// <inheritdoc cref="IComparable{T}.CompareTo" />
+        public int CompareTo(Pressure other)
+        {
+            return _pascal.CompareTo(other._pascal);
+        }
+
         /// <summary>
-        /// Creates a string representation of this object, in hPa.
+        /// Returns the string representation of this pressure, in hPa
         /// </summary>
+        /// <returns>String representation of this pressure</returns>
         public override string ToString()
         {
-            return string.Format(CultureInfo.CurrentCulture, "{0:F2}hPa", Hectopascal);
+            return String.Format(CultureInfo.CurrentCulture, "{0:F1} hPa", Hectopascal);
         }
 
         /// <summary>
-        /// Equality member
+        /// Returns the string representation of this pressure, with the given format string and using the current culture.
+        /// For valid formatting arguments, see <see cref="ToString(string, IFormatProvider)"/>
         /// </summary>
-        public bool Equals(Pressure other)
+        /// <param name="formatArgs">Format string</param>
+        /// <returns>String representation</returns>
+        public string ToString(string formatArgs)
         {
-            return _pascal.Equals(other._pascal);
+            return ToString(formatArgs, CultureInfo.CurrentCulture);
         }
 
         /// <summary>
-        /// Equality member
+        /// Returns the string representation of this pressure, with the given format string and using the given culture.
+        /// Valid format specifiers are:
+        /// PA: Pascal
+        /// MBAR: Millibar
+        /// KPA: Kilopascal
+        /// HPA: Hectopascal
+        /// INHG: Inch of mercury
+        /// MMHG: Millimeter of mercury
+        /// An extra number defines the number of decimal digits to use (default 1)
+        /// <example>
+        /// <code>
+        /// var s = p.ToString("PA2"); -> "101325.01 Pa"
+        /// var s = p.ToString("HPA2"); --> "1013.25 hPa"
+        /// </code>
+        /// </example>
         /// </summary>
-        public override bool Equals(object obj)
+        /// <param name="formatArgs">Format string</param>
+        /// <param name="culture">Culture to use. Affects the format of the number.</param>
+        /// <returns>String representation</returns>
+        public string ToString(string formatArgs, IFormatProvider culture)
```

Comment:
I don't feel strongly about it but I think until we fully design how this formatter should work we should remove it. I'm suspecting this currently won't cover all cases. Some people will want to put space between number and unit, some will prefer to display it slightly differently, some will have translated unit name or use different alphabel (not sure about it, just guessing). I'd rather throw on ToString or return just .Pascal.ToString